### PR TITLE
Upgrade testing support

### PR DESCRIPTION
**Pull request by psss**
Initial brainstorming of possible approaches.
Relates to #694.

### CODE REVIEW

File: tests/execute/upgrade/data/multiple-steps.fmf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+discover:
+    how: fmf
+prepare-1:
+    how: ansible
+    playbook: old.yaml
+execute-1:
+    how: tmt
+prepare-2:
+    how: ansible
+    playbook: old.yaml
```
Comment (pvalena) [inline tests/execute/upgrade/data/multiple-steps.fmf:10]:
```suggestion
    playbook: new.yaml
```

**Comment by pvalena** [comment]
One Idea I wanted to pitch in, inspired by Ruby (new-ish) feature `Guilds` (for handling process parallelism). We can call them Groups f.e., for simplicity (and because we don't handle memory locking, yet).

Basically, you'd keep everything as-is (Plan, sub-plan notation etc.), but with addition of Plan group, which would somehow link plans together.

It can be implemented in various ways.

E.g. by saying the step is "shared" across sub-plans.
```yaml
discover:
    how: fmf

execute:
    how: tmt

/test-old:
    provision:
        group: True
    prepare:
        how: ansible
        playbook: old.yaml

/test-new:
    provision:
        group: True
    prepare:
        how: ansible
        playbook: new.yaml
```

Or by hard-coding `provision` as a shared step, and simply saying which sub-plans need to be groupped:
```yaml
discover:
    how: fmf

execute:
    how: tmt

/test-old:
    group: True
    prepare:
        how: ansible
        playbook: old.yaml

/test-new:
    group: True
    prepare:
        how: ansible
        playbook: new.yaml
```

Some other options:

- Replace `group` with `shared`, as an adjectivum.
- Instead of `True` as a group value, you could have a string, or number (i.e. `group: 1`).
- Inverted value, for property `isolation` (i.e. `isolate: False` instead of `group: True`).

**Comment by lukaszachy** [comment]
@pvalena @psss Is this still relevant? If so, it seems to share ideas with to #1142

**Comment by lukaszachy** [comment]
@psss Since we have 'upgrade' plugin for execute released - is this still relevant?

**Pull request closed without merging**